src/components/overlays/AlertDialog/AlertDialog.stories.tsx

```diff
@@ -82,3 +89,28 @@ const Template = (props) => {
 
 export const Default = Template.bind({});
 Default.args = {};
+
+const ApiTemplate = (args) => {
+  const api = useAlertDialogApi();
+
+  return (
+    <Button
+      {...args}
+      onPress={() => {
+        const { promise } = api.open({
```

Comment:
Doesn't look like a convenient way to use API. It's better to use primitive properties like `onConfirm` and `onDismiss` than that.
The better way would be to return the promise directly from the `open` method. Anyway, I'm not sure we need the `close` method here. Do you know any case?